### PR TITLE
Remove TRUNCATE PARTITION as non-supported

### LINKED ISSUE #12311: ALTER TABLE t TRUNCATE PARTITION is not mentioned as supported.
## Change Request

Please answer the following questions before submitting your issue. Thanks!

1. Describe what you find is inappropriate or missing in the existing docs.
In https://docs.pingcap.com/tidb/dev/mysql-compatibility#ddl in the list after "Table Partitioning also supports ADD, DROP, and TRUNCATE operations. Other partition operations are ignored. The following Table Partition syntaxes are not supported:" TRUNCATE is mentioned as not supported, this is wrong, TRUNCATE PARTITION is supported.

2. Describe your suggestion or addition.
Change:
"{CHECK|TRUNCATE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION"
to
"{CHECK|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION"

3. Provide some reference materials (such as documents and websites) if you could.
Read the paragraph in front, where it says "Table Partitioning also supports ADD, DROP, and TRUNCATE operations."

### PR DESCRIPTION
**Pull request by mjonss**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): close #12311

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: mysql-compatibility.md [hunk 1]
```diff
@@ -149,12 +149,13 @@ In TiDB, all supported DDL changes are performed online. Compared with DDL opera
 * The `ALGORITHM={INSTANT,INPLACE,COPY}` syntax functions only as an assertion in TiDB, and does not modify the `ALTER` algorithm. See [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md) for further details.
 * Adding/Dropping the primary key of the `CLUSTERED` type is unsupported. For more details about the primary key of the `CLUSTERED` type, refer to [clustered index](/clustered-indexes.md).
 * Different types of indexes (`HASH|BTREE|RTREE|FULLTEXT`) are not supported, and will be parsed and ignored when specified.
-* Table Partitioning supports `HASH`, `RANGE`, and `LIST` partitioning types. For the unsupported partition type, the `Warning: Unsupported partition type %s, treat as normal table` error might be output, where `%s` is a specific partition type.
+* Table Partitioning supports `HASH`, `RANGE`, and `LIST` partitioning types. For the unsupported partition type, the `Warning: Unsupported partition type %s, treat as normal table` warning will given, where `%s` is a specific partition type.
```
Comment (qiancai) [inline mysql-compatibility.md:152]:
```suggestion
* Table Partitioning supports `HASH`, `RANGE`, and `LIST` partitioning types. For the unsupported partition type, the `Warning: Unsupported partition type %s, treat as normal table` warning will be returned, where `%s` is a specific partition type.
```

File: mysql-compatibility.md [hunk 2]
```diff
@@ -149,12 +149,13 @@ In TiDB, all supported DDL changes are performed online. Compared with DDL opera
...
 * Different types of indexes (`HASH|BTREE|RTREE|FULLTEXT`) are not supported, and will be parsed and ignored when specified.
-* Table Partitioning supports `HASH`, `RANGE`, and `LIST` partitioning types. For the unsupported partition type, the `Warning: Unsupported partition type %s, treat as normal table` error might be output, where `%s` is a specific partition type.
+* Table Partitioning supports `HASH`, `RANGE`, and `LIST` partitioning types. For the unsupported partition type, the `Warning: Unsupported partition type %s, treat as normal table` warning will be returned, where `%s` is a specific partition type.
 * Table Partitioning also supports `ADD`, `DROP`, and `TRUNCATE` operations. Other partition operations are ignored. The following Table Partition syntaxes are not supported:
     - `PARTITION BY KEY`
     - `PARTITION BY LINEAR KEY`
     - `SUBPARTITION`
-    - `{CHECK|TRUNCATE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION`
+* And the following `ALTER TABLE` commands are not supported:
```
Comment (qiancai) [inline mysql-compatibility.md:157]:
```suggestion
* The following `ALTER TABLE` commands are not supported:
```

**Review by TonsnakeLin — APPROVED** [review]

**Comment by yiwen92** [comment]
pls merge into master branch

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.4 label because the v6.4 docs have been archived at https://docs-archive.pingcap.com/tidb/v6.4 and will no longer receive new updates.

**Comment by qiancai** [comment]
Hi @mjonss, as the master docs had been updated and the dev doc for the TRUNCATE PARTITION is now correct, can we close this PR? I've created another PR for docs before v7.0.  https://github.com/pingcap/docs/pull/13439. PTAL. Thanks.

**Pull request closed without merging**